server/v2/cometbft/client/grpc/cmtservice/pagination.go

```diff
@@ -0,0 +1,44 @@
+package cmtservice
+
+import (
+	math "math"
+
+	codes "google.golang.org/grpc/codes"
+	status "google.golang.org/grpc/status"
+)
+
+// DefaultPage is the default `page` number for queries.
+// If the `page` number is not supplied, `DefaultPage` will be used.
+const DefaultPage = 1
```

Comment:
pagination seems like a generic thing, any reason we cant put it into the api pkg at the in server/v2?

Reply:
Or we put a variation of it in core. This is to avoid sdk imports of pagination in modules. Thoughts?

Reply:
There's also a proto generated file that contains PageRequest and some other. I think this is also extracted from comet, so if we want to extend this same pagination style everywhere else then we should make it like you said

Reply:
can we use the api module here?

Reply:
I think we talked about this on a call, but that would mean we use a pulsar type. Also, those types use codec/any for pubkeys, so we are still going to have a dependency on the sdk

Reply:
so i think this is fine for now, we should get this merged, but we will need to do another refactor here because we dont want to use the pulsar types. Im not sure where pagination should live to avoid the imports, maybe you can dive into this?

Reply:
I'm a bit confused, should we not use pulsar on this pr? I started without it but then we talked about using it so I started to use it and added a bunch of utils to make that possible here: [use cosmossdk.io/api](https://github.com/cosmos/cosmos-sdk/pull/19744/commits/8b78f56e46d27a7ec09dee57d2eb456024b90082)

Reply:
we should be able to depend on server v2 here, we can put it in there. the issue is we are adding multiple versions of pagination now, which makes client world harder